build/src/lib.rs

```diff
@@ -148,52 +281,59 @@ pub struct BinaryBuilder {
     fallback_to_build_dir: bool,
     build_dir: Option<PathBuf>,
     outputs: Vec<PathBuf>,
+    transform_path_fn: Option<Box<dyn TransformPathFn>>,
+    transform_data_fn: Option<Box<dyn TransformDataFn<Cursor<Vec<u8>>>>>,
 }
 
 impl BinaryBuilder {
+    /// ```plain
+    /// github release resource --[fn transform]--> $cache_dir --[fn postprocess]--> $OUT_DIR/
+    /// ```
+    ///
+    /// ```plain
+    /// github release resource --[fn transform]--> $OUT_DIR --[fn postprocess]--> $OUT_DIR/
+    /// ```
+    ///
     /// Acquires the rule and tokenizer binaries for one language by:
     /// - Trying to download them from their distribution source (or load them local cache).
     /// - If they are not found (i. e. a dev version of nlprule is used) and `fallback_to_build_dir`is true
     /// downloads the latest build directory  and builds the binaries from it.
     /// This can still fail if the dev version is sufficiently outdated for the latest build dir.
-    /// In that case, the user can't be helped.
-    fn build_language(&mut self, lang_code: &str) {
-        let tokenizer_out = self.out_dir.join(tokenizer_filename(lang_code));
-        let rules_out = self.out_dir.join(rules_filename(lang_code));
+    /// In that case, the user is encouraged to update to a release or a newer git sha.
+    fn build_language(&mut self, lang_code: &str) -> Result<()> {
```

Comment:
In `build_language` we need to also call the transform function when falling back to building from the build directory. It would also be great if you could fix the `.expect`s here too (e. g. L350) but I can do that later too.

Reply:
I think I covered most of them for now.